lib/index.js

```diff
@@ -25,12 +25,17 @@ exports.register = (server, options, next) => {
     options.redisClient.multi();
     options.redisClient.set(key, 0, 'EX', rate.window, 'NX'); // if key not found, insert key:0 with window seconds expiry
     options.redisClient.incr(key);
+    options.redisClient.ttl(key);
+
     return options.redisClient.exec()
     .then((results) => {
-      rate.current = results[1]; // result of 2nd redis instruction
+      const remaining = rate.limit - results[1]; // result of 2nd redis instruction
+
+      rate.remaining = remaining > 0 ? remaining : 0;
+      rate.reset = Math.floor(new Date() / 1000) + results[2];
```

Comment:
can you do `Math.floor((new Date()).getTime() / 1000)` so that we're explicit that we're taking the milliseconds and dividing it (as opposes to taking a date object and dividing it). it works the same, but it makes the code a bit more clear.